library/x509_crt.c

```diff
@@ -620,231 +879,559 @@ static int x509_get_ext_key_usage( unsigned char **p,
  *
  * NOTE: we only parse and use dNSName at this point.
  */
-static int x509_get_subject_alt_name( unsigned char **p,
+static int x509_get_subject_alt_name( unsigned char *p,
                                       const unsigned char *end,
                                       mbedtls_x509_sequence *subject_alt_name )
+{
+    return( mbedtls_asn1_traverse_sequence_of( &p, end,
+                                               MBEDTLS_ASN1_TAG_CLASS_MASK,
+                                               MBEDTLS_ASN1_CONTEXT_SPECIFIC,
+                                               MBEDTLS_ASN1_TAG_VALUE_MASK,
+                                               2 /* SubjectAlt DNS */,
```

Comment:
I agree in regards to macros, but don't in regards to not putting comments like this - I think it's easy to read this way, and prefer to keep it.

I'll check if @RonEld introduced defined constants for the DNS types in #2530, and introduce them here otherwise (which will also render the inline comment superfluous, but the general point about their use remains).